File: tsl_logger_py/examples/value_logger.py
# Copyright 2024 Simon Sagmeister
from tsl_logger_py import ValueLogger


logger1 = ValueLogger()

# You can just log any numeric value
logger1.log("int_test", 2)
logger1.log("float_test", 3.0)

# You can even log lists or dicts
vec = [
    5.0,
    6.0,
    7.0,
]  # Avoid mixing floats and ints in a vectors. Even though it is possible.
map = {"signal_a": 1, "signal_b": 2}
logger1.log("Testvector", vec)
logger1.log("Testmap", map)


# WARNING: Everything that is logged with the logger.log() function will be logged as 64 bit datatype

# This is due to to the dynamic sizing of python numeric types. So its not possible to determine
# which integer/floating type to use.
# However, you can explicitly specify a datatype that will be used for logging your value
logger1.log_float64("float64_test", 6.0)
logger1.log_float32("float32_test", 6.0)
logger1.log_uint64("uint64_test", 5)
logger1.log_int64("int64_test", 5)
logger1.log_uint32("uint32_test", 5)
logger1.log_int32("int32_test", 5)
logger1.log_uint16("uint16_test", 5)
logger1.log_int16("int16_test", 5)
logger1.log_uint8("uint8_test", 5)
logger1.log_int8("int8_test", 5)


# You can even log the values contained in another logger
# Note: This will copy the data over. For a faster way of combining loggers, see the composition
# example
logger2 = ValueLogger()
logger2.log("logger1_data", logger1.get_data())

print("We logged the following signals in Logger 1:")
print(type(logger1.get_data()))
for name, value in logger1.get_data().items():
    print("- " + name + f": {value}")

print("\n\n\n")

print("We logged the following signals in Logger 2:")
for name, value in logger2.get_data().items():
    print("- " + name + f": {value}")

File: tsl_logger_cpp/examples/value_logger.cpp
// Copyright 2024 Simon Sagmeister

#include "tsl_logger_cpp/value_logger.hpp"

#include <iostream>

#include "tsl_logger_cpp/base.hpp"
int main()
{
  tam::tsl::ValueLogger logger1;

  // You can just log any numeric value
  logger1.log("double_test", 1.0);
  logger1.log("int_test", 2);
  logger1.log("float_test", 3.0f);
  std::int64_t int64_val = 4;
  logger1.log("int64_test", int64_val);

  // You can even log a whole range of standard containers
  std::vector<float> vec = {5.0, 6, 7};
  std::array<double, 3> arr = {5.0, 6, 7};
  std::unordered_map<std::string, int> map = {{"sigal_a", 1}, {"signal_b", 2}};
  logger1.log("Testvector", vec);
  logger1.log("Testarray", arr);
  logger1.log("Testmap", map);

  // You can even log the values contained in another logger
  // Note: This will copy the data over. For a faster way of combining loggers, see the composition
  // example
  tam::tsl::ValueLogger logger2;
  logger2.log("logger1_data", logger1.get_data());

  std::cout << "We logged the following signals in Logger 1:" << std::endl;
  for (auto const & signal : logger1.get_data()) {
    std::cout << "- " << signal.first << std::endl;
  }
  std::cout << std::endl << std::endl << std::endl;

  std::cout << "We logged the following signals in Logger 2:" << std::endl;
  for (auto const & signal : logger2.get_data()) {
    std::cout << "- " << signal.first << std::endl;
  }

  return 0;
}
